specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,981 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List Policies in a Resource Group": {
+            "$ref": "./examples/WafListManagedRuleSets.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "WebApplicationFirewallPolicy": {
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of the web application firewall policy.",
+          "$ref": "#/definitions/WebApplicationFirewallPolicyProperties"
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        },
+        "sku": {
+          "description": "The pricing tier of web application firewall policy. Defaults to Classic_AzureFrontDoor if not specified.",
+          "$ref": "#/definitions/Sku"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyProperties": {
+      "description": "Defines web application firewall policy properties.",
+      "properties": {
+        "policySettings": {
+          "description": "Describes settings for the policy.",
+          "$ref": "#/definitions/PolicySettings"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy.",
+          "$ref": "#/definitions/CustomRuleList"
+        },
+        "managedRules": {
+          "description": "Describes managed rules inside the policy.",
+          "$ref": "#/definitions/ManagedRuleSetList"
+        },
+        "frontendEndpointLinks": {
+          "description": "Describes Frontend Endpoints associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FrontendEndpointLink"
+          }
+        },
+        "routingRuleLinks": {
+          "description": "Describes Routing Rules associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/RoutingRuleLink"
+          }
+        },
+        "securityPolicyLinks": {
+          "description": "Describes Security Policy associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/SecurityPolicyLink"
+          }
+        },
+        "provisioningState": {
```

Comment:
>provisioningState [](http://example.com/codeflow?start=9&length=17)

define as enum and have the ARM's definition for it with min values Succeeded, Canceled, Failed

Reply:
Do you have an undefined list of provisioning states? Usually those are fixed and you add them here, see this example: [https://github.com/Azure/azure-rest-api-specs/blob/b8fc2b4ff626aab886f929e49926c6e836548e01/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json#L5044](https://github.com/Azure/azure-rest-api-specs/blob/b8fc2b4ff626aab886f929e49926c6e836548e01/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json#L5044)

Reply:
Done